### DISCUSSION TITLE
Using next.config.js redirects() when having dynamic routes

### DISCUSSION THREAD

**Discussion by waptik** [vercel/next.js#19457] [Help] [answered]
Hi! I kindaa need help with next.config.js redirects() along side dynamic routes.

In my project, i have have dynamic route at the top level(pages/[creator]/...) and i would like to use redirects inside next.config.js.

Here is my use case:
1. pages/[creator]/... => user profile pages
2. /discord => a path set inside next.js.config that will redirect anyone who visits that page/path to my discord server.

This is what i tried so far:

I -

```js
// next.config.js
...
redirects: async function redirect() {
    return [
        {
            source: "/discord",
            destination: "https://discord.gg/letsgothere",
        },
    ]
}
...
```

II -

```js
// next.config.js
...
redirects: async function redirect() {
    return [
        {
            source: "/discord",
            destination: "https://discord.gg/letsgothere",
            permanent: true,
        },
    ]
}
...
```

Both I & II don't redirect to discord but instead go 1([creator]/...).

I might be doing it the wrong way or maybe this approach isn't supported.
Any advice is welcomed.

Thanks

**Comment by Tomburgs** [comment]
There doesn't seem to be anything wrong with how you're doing it.

Are you using Next.js 9.5 or later?

Also, how are you redirecting to this route? You cannot be using the `next/link` component for this.

**Reply by waptik** [reply]
Hi @Tomburgs, thanks for replying!

I'm using next.js 10.0.2

I just type `http://localhost:3000/discord` in my browser's address bar and i get sent to profile page instead of discord server

**Comment by waptik** [comment, marked as answer]
So i fixed it. As you can see in both `I & II`, i was missing an `s` inside the async function `redirect`. So now it's no longer `async function redirect()` but `async function redirects()`.

> Hi! I kindaa need help with next.config.js redirects() along side dynamic routes.
> 
> In my project, i have have dynamic route at the top level(pages/[creator]/...) and i would like to use redirects inside next.config.js.
> 
> Here is my use case:
> 
> 1. pages/[creator]/... => user profile pages
> 2. /discord => a path set inside next.js.config that will redirect anyone who visits that page/path to my discord server.
> 
> This is what i tried so far:
> 
> I -
> 
> ```js
> // next.config.js
> ...
> redirects: async function redirect() {
>     return [
>         {
>             source: "/discord",
>             destination: "https://discord.gg/letsgothere",
>         },
>     ]
> }
> ...
> ```
> 
> II -
> 
> ```js
> // next.config.js
> ...
> redirects: async function redirect() {
>     return [
>         {
>             source: "/discord",
>             destination: "https://discord.gg/letsgothere",
>             permanent: true,
>         },
>     ]
> }
> ...
> ```
> 
> Both I & II don't redirect to discord but instead go 1([creator]/...).
> 
> I might be doing it the wrong way or maybe this approach isn't supported.
> Any advice is welcomed.
> 
> Thanks
